app.ts

```diff
@@ -0,0 +1,143 @@
+//Loads the table to the screen
+window.onload = () => GetColumnNames();
+
+//Variable declarations
+let totalRecordCount;
+let columnNames: string[];
+let records: any[];
+let from: number = 1;
+let to: number = 30;
+
+//Function to retrieve total amount of records
+function GetNumberOfRecords() {
+  $.ajax({
+    url: "/recordCount",
+    type: "GET",
+    timeout: 12000,
+    dataType: "text"
+  })
+    .done(function(responseText: any) {
+      totalRecordCount = responseText;
+    })
+    .fail(function() {
+      alert("Could not retrieve total record number");
+    });
+}
+
+//function to retrieve column names
+function GetColumnNames() {
+  $.ajax({
+    url: "/columns",
+    type: "GET",
+    timeout: 12000,
+    dataType: "JSON"
+  })
+    .done(function(responseJSON: any) {
+      columnNames = responseJSON;
+      Sync();
+    })
+    .fail(function() {
+      alert("Could not display column names");
+    });
+}
+
+//Function to retrieve the record data
+function GetActualRecords() {
+  $.ajax({
+    url: "/records?from=" + from + "&to=" + to,
+    timeout: 12000,
+    dataType: "JSON"
+  })
+    .done(function(responseRecords: any) {
+      records = responseRecords;
+      NextSync();
+    })
+    .fail(function() {
+      alert("No available records for your selection");
+    });
+}
+
+//Function to sync API calls
+function Sync() {
+  GetActualRecords();
+}
+
+//Function to sync API calls
+function NextSync() {
+  BuildTable();
+}
+
+//Function builds the table and populates with data from the API calls
+function BuildTable() {
+  RemoveTable();
+
+  const body = document.getElementsByTagName("body")[0];
+
+  const ourTable = document.createElement("table");
+  ourTable.setAttribute("id", "Table");
+
+  const thead = document.createElement("thead");
+  ourTable.appendChild(thead);
+
+  //Column names are created for the table
+  for (let c = 0; c < columnNames.length; c++) {
```

Comment:
Let's make c a const, since at the end of every iterative loop it gets re-assigned.
```ts
   for (const c = 0; c < columnNames.length; c++) {
```
It is a bit trippy at first, but for the sake of convention.